lib/rubygems/ext/cargo_builder/link_flag_converter.rb

```diff
@@ -3,20 +3,24 @@
 class Gem::Ext::CargoBuilder < Gem::Ext::Builder
   # Converts Ruby link flags into something cargo understands
   class LinkFlagConverter
+    FILTERED_PATTERNS = [
+      /compress-debug-sections/, # Not supported by all linkers, and not required for Rust
+    ].freeze
+
     def self.convert(arg)
+      return [] if FILTERED_PATTERNS.any? {|p| p.match?(arg) }
+
       case arg.chomp
       when /^-L\s*(.+)$/
         ["-L", "native=#{$1}"]
       when /^--library=(\w+\S+)$/, /^-l\s*(\w+\S+)$/
         ["-l", $1]
-      when /^-l\s*:lib(\S+).a$/
-        ["-l", "static=#{$1}"]
-      when /^-l\s*:lib(\S+).(so|dylib|dll)$/
-        ["-l", "dylib=#{$1}"]
+      when /^-l\s*([^:\s])+/ # -lfoo, but not -l:libfoo.a
+        ["-l", $1]
```

Comment:
When matching `"-lfoo"`, `$1` will be the last letter `"o"` only.
Is this expected?

And `when /^-l\s*(\w+\S+)$/` covers `"-lfoo"`, so this `when` clause only matches `-l` with a punctuation, such as `"-l-foo"`.
Is this expected?